Tickets/dm 15755

MichelleGower opened the pull request.

MichelleGower committed:
Add command line option for transfer URL prefix

MichelleGower committed:
Add retries and some better error handling

mxk62 commented on bin.src/dbb_save_ats_raw.py:
```diff
@@ -63,14 +63,28 @@ def parse_args(argv=None):
+    optional.add_argument("--dryrun", action="store_true", dest="dryrun", required=False,
+                          help="If set, does not actually transfer file")
+
+    parser._action_groups.append(optional)
```
While I understand the desire to display the description of required arguments first, I'm not sure if it is worth the trouble. Using `argparse` internals indicates that it's a hack rather than a solution. However, I don't mind you keeping it that way. Just put a comment explaining why you use this "magic".

MichelleGower replied:
Replaced `argparse` internals with normal argparse group.

mxk62 commented on bin.src/dbb_save_ats_raw.py:
```diff
@@ -214,13 +228,61 @@ def create_transfer_cmd(filename, dest, uuid_str):
+    trans_opts : `dict`
+        Options for the transfer command (e.g., dest http url prefix)
+    """
+
```
Remove the blank line between the docstring and the first statement.

MichelleGower committed:
Fix docstring formatting

Removed blank line after docstring, added period to end of parameter
description lines.

MichelleGower committed:
Remove use of argparse internals

Replaced use of argparse internals (_action_groups) to use
normal group function.
